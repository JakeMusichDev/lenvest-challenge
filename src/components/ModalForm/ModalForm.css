.modal-form-container {
  grid-row: 4/5;
  grid-column: 1/5;
  padding: 2%;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
}

@media only screen and (max-width: 992px) {
  .form-text-input {
    width: 40%;
    margin-bottom: 5px;
    text-align: center;
  }
  
  .form-text-submit {
    width: 40%;
    border-radius: 25px;
    border: 1px solid rgba(42, 21, 233, 0.6);
    color: rgba(42, 21, 233, 0.8);
    cursor: pointer;
    transition: 0.3s all;
  }
}

@media only screen and (min-width: 992px) {
  .form-text-input {
    width: 20%;
    margin-bottom: 5px;
    text-align: center;
  }
  
  .form-text-submit {
    width: 20%;
    border-radius: 25px;
    border: 1px solid rgba(42, 21, 233, 0.6);
    color: rgba(42, 21, 233, 0.8);
    cursor: pointer;
    transition: 0.3s all;
  }
}

.list-item-btn:hover {
  border: 1px solid rgba(42, 21, 233, 1);
  color: rgba(42, 21, 233, 1);
}

