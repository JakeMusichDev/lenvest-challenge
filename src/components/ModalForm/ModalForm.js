import React, { Component } from 'react'

export default class ModalForm extends Component {
  constructor(props) {
    super(props)

    this.state = {inputValue: ""}
  }

  handleChange = (e) => {
    this.setState({inputValue: e.target.value})
  }

  handleSubmit = e => {
    this.props.setFormValue(this.state.inputValue)
    e.preventDefault()
  }

  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>Investment Amount $</label>
        <input type="text" onChange={this.handleChange}/>
        <input type="submit" value="Submit" />
      </form>
    )
  }
}