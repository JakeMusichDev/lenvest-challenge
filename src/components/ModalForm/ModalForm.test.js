import React from 'react'
import {shallow} from 'enzyme'
import ModalForm from './ModalForm'

describe('ModalForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ModalForm />);
  });

  it('renders form', () => {
    expect(wrapper.find("form").length).toEqual(1)
  })

  it('renders two form input', () => {
    expect(wrapper.find('input').length).toEqual(2)
  })

  test('pass a selected value to the onChange handler', () => {
    // const value = '2';
    // const onChange = jest.fn();
    // const wrapper = shallow(
    //     <ModalForm onChange={onChange} />
    // );

    // expect(wrapper).toMatchSnapshot();

    // wrapper.find('input').simulate('change', {
    //   target: { value },
    // });

    // expect(onChange).toBeCalledWith(value);
});
})