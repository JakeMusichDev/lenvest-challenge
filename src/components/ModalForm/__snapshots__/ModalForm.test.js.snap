// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalForm pass a selected value to the onChange handler 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalForm
    onChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Investment Amount $
</label>,
        <input
          onChange={[Function]}
          type="text"
/>,
        <input
          type="submit"
          value="Submit"
/>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Investment Amount $",
        },
        "ref": null,
        "rendered": "Investment Amount $",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
          "value": "Submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Investment Amount $
</label>,
          <input
            onChange={[Function]}
            type="text"
/>,
          <input
            type="submit"
            value="Submit"
/>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Investment Amount $",
          },
          "ref": null,
          "rendered": "Investment Amount $",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "Submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
